core/crypto/sub.mk

```diff
@@ -42,4 +42,5 @@ srcs-$(CFG_CRYPTO_ECB) += sm4-ecb.c
 srcs-$(CFG_CRYPTO_CBC) += sm4-cbc.c
 srcs-$(CFG_CRYPTO_CTR) += sm4-ctr.c
 srcs-$(CFG_CRYPTO_XTS) += sm4-xts.c
+srcs-$(CFG_CRYPTO_HW_PBKDF2) += pbkdf2_hw.c
```

Comment:
I don't think `pbkdf2_hw.c` or `pbkdf2_support.h` is needed at all. The hardware implementation can just implement the `tee_cryp_pbkdf2()`

Reply:
Is that mean:
If the two files are not needed, the hardware implementation also put in `tee_cryp_pbkdf2.c` and use CFG_CRYPTO_HW_PBKDF2 to distinguish?

Reply:
No, a hardware non-stubbed implementation is provided in some other file.
It doesn't make sense to use a stubbed hardware implementation when a working software version is available.

Reply:
The hardware implementation will be different from the current software, we will use different hash algorithm etc.
So, I think the  `pbkdf2_hw.c` and `pbkdf2_support.h` is necessary, then specific implementation of `hw_pbkdf2()` function will be added in `core/ drivers/crypto` @jenswi-linaro

Reply:
Do you mean that the same input will yield different results? I don't see how a stubbed implementation helps against that.

Reply:
> Do you mean that the same input will yield different results? I don't see how a stubbed implementation helps against that.

I'm sorry for misunderstanding your previous comment. Now, I got it and the PR is ready for another round of review.